File: 1302-delete-characters-to-make-fancy-string/solution.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int n = s.size();
        if (n < 3) return s;
        int j = 2;
        
        for (int i = 2; i < n; i++) {
            if (s[i] != s[j-1] || s[i] != s[j-2]) {
                // not equal to last 2 chars
                s[j++] = s[i];
            }
        }

        s.resize(j);
        return s;
        
    }
};

File: 1302-delete-characters-to-make-fancy-string/solution.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        n = len(s)
        if n < 3:
            return s
        
        # use stack approach. no three consecutive chars are equal.
        # create a stack and dont push if dq.count(num) is 3.
        # at the end, convert the stack to string

        dq = []

        for c in s:
            if len(dq) >= 2 and dq[-1] == c and dq[-2] == c:
                continue
            
            dq.append(c)
        
        return "".join(dq)
